be/src/olap/rowset/segment_v2/segment_writer.h

```diff
@@ -60,6 +60,8 @@ class SegmentWriter {
 
     uint64_t estimate_segment_size();
 
+    bool reach_capacity();
```

Comment:
I prefer to have a function named `num_rows_written()` and move the logic of when to flush segment to BetaRowsetWriter.